mrjob/conf.py

```diff
@@ -68,8 +67,8 @@ def default_options(self):
     def validated_options(self, opts, error_fmt):
         unrecognized_opts = set(opts) - self.ALLOWED_KEYS
         if unrecognized_opts:
-            log.warn(error_fmt % ', '.join(sorted(unrecognized_opts)))
-            return dict((k, v) for k, v in opts.iteritems()
+            log.warning(error_fmt % ', '.join(sorted(unrecognized_opts)))
```

Comment:
pylint would give you a warning here: non-lazy log
The log functions take two arguments, a %-style template, and values for the template.
It defers actually templating the stuff until it decides to emit the logline.

P4